routers/repo/branch.go

```diff
@@ -30,3 +31,55 @@ func Branches(ctx *context.Context) {
 	ctx.Data["Branches"] = brs
 	ctx.HTML(200, tplBranch)
 }
+
+// CreateBranch creates new branch in repository
+func CreateBranch(ctx *context.Context, form auth.NewBranchForm) {
+	if !ctx.Repo.CanCreateBranch() {
+		ctx.Handle(404, "CreateBranch", nil)
+		return
+	}
+
+	if ctx.HasError() {
+		ctx.Flash.Error(ctx.GetErrMsg())
+		ctx.Redirect(ctx.Repo.RepoLink + "/src/" + ctx.Repo.BranchName)
+		return
+	}
+
+	branches, err := ctx.Repo.Repository.GetBranches()
+	if err != nil {
+		ctx.Handle(500, "GetBranches", err)
+		return
+	}
+
+	for _, branch := range branches {
+		if branch.Name == form.NewBranchName {
+			ctx.Flash.Error(ctx.Tr("repo.branch.branch_already_exists", branch.Name))
+			ctx.Redirect(ctx.Repo.RepoLink + "/src/" + ctx.Repo.BranchName)
+			return
+		} else if (len(branch.Name) < len(form.NewBranchName) && branch.Name+"/" == form.NewBranchName[0:len(branch.Name)+1]) ||
```

Comment:
Could we move this condition to a helper function, something like `BranchNamesConflict(name1, name2)`? It could be useful in other places, and would make the code more readable IMO.

Reply:
@ethantkoenig I have no idea to where to move this functions

Reply:
`models.(*Repository).BranchNameConflict(form.NewBranchName)` perhaps?